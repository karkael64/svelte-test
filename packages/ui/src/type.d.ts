export { Button } from "./stories/atom/button/button";
export { Header } from "./stories/example/header/header";
export { Page } from "./stories/example/page/page";
