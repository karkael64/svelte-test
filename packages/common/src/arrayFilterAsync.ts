export default async function arrayFilterAsync<Item extends any>(
  list: Item[],
  each: (value: Item, index: number, array: Item[]) => Promise<any>,
  thisArgs?: any
) {
  const proms = [];
  for (const key in list) {
    proms.push([
      each.call(thisArgs, list[key], parseInt(key), list),
      list[key],
    ]);
  }
  const replies = await Promise.all(proms);

  return replies.filter((reply) => reply[0]).map((reply) => reply[1]) as Item[];
}
