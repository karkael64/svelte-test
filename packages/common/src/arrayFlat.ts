export default function arrayFlat<Type>(list: Type[][], depth = 1): Type[] {
  if (depth < 1) return list as any;
  return arrayFlat(list, depth - 1).reduce((acc, val) => {
    Array.isArray(val) ? acc.concat(val) : acc.push(val);
    return acc;
  }, [] as Type[]);
}

declare global {
  interface Array<T> {
    flat<Type>(list: Type[][], depth?: number): Type[];
  }
}

if (Array.prototype.flat === undefined) {
  Object.defineProperty(Array.prototype, "flat", {
    enumerable: false,
    configurable: true,
    writable: true,
    value: function (list: any, depth?: number) {
      return arrayFlat(list, depth);
    },
  });
}
