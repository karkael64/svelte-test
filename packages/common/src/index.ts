export { default as arrayFilterAsync } from "./arrayFilterAsync";
export { default as arrayFlat } from "./arrayFlat";
export { default as arrayGroupBy } from "./arrayGroupBy";
export { default as arrayMapAsync } from "./arrayMapAsync";
export { default as arrayReduceAsync } from "./arrayReduceAsync";
export { ellipsis, ellipsisQuoted, escapedEllipsis } from "./ellipsis";
export { default as injectImports } from "./injectImports";
export { objectHasPorperty, objectHasProperties } from "./objectHasProperty";
export { default as objectEntries } from "./objectEntries";
export { default as objectMap } from "./objectMap";
export { default as objectSort } from "./objectSort";
export { isEmail, isPassword, isUsername } from "./regex";
export { default as stringPascalCase } from "./stringPascalCase";
export {
  stringShouldMatch,
  shouldBeVariableName,
  shouldBeResolvePath,
} from "./syntax";
