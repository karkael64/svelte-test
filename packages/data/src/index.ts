import { Router } from "express";
import { routes } from "./routes";
export { default as graphql } from "./resolvers";

export const router = Router();
routes.forEach((routeDef) => {
  router[routeDef.method](routeDef.path, routeDef.handler);
});
