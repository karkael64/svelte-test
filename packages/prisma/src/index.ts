import dotenv from "dotenv";
dotenv.config();

import { PrismaClient, PrismaPromise } from "../client";

export const prisma = new PrismaClient();

export const dbTransaction = (
  promises: PrismaPromise<any>[]
): Promise<PromiseConstructor[]> => prisma.$transaction(promises);
