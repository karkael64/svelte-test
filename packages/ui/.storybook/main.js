module.exports = {
  stories: ["../src/**/*.stories.mdx", "../src/**/*.stories.@(js|ts|svelte)"],
  addons: [
    "@storybook/addon-links",
    "@storybook/addon-essentials",
    "@storybook/addon-svelte-csf",
    "@storybook/addon-actions",
    "@storybook/addon-postcss",
  ],
  svelteOptions: {
    preprocess: require("svelte-preprocess")(),
  },
};
