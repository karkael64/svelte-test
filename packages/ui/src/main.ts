export { default as Button } from "./stories/atom/button/Button.svelte";
export { default as Header } from "./stories/example/header/Header.svelte";
export { default as Page } from "./stories/example/page/Page.svelte";
