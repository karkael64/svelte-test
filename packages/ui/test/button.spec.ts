import { fireEvent } from "@testing-library/svelte";
import { Button } from "../src/main";
import type { Button as ButtonType } from "../src/main.d";
import { renderWithEvents } from "./renderWithEvents";

describe("Button", () => {
  it("Should instanciate with default values", async () => {
    const label = "inner text";
    const { getByText, component } = renderWithEvents<ButtonType>(Button, {
      label,
    });

    const button = getByText(label);
    expect(button).toBeInTheDocument();

    expect(button).toHaveClass("storybook-button");
    expect(button).toHaveClass("storybook-button--medium");
    expect(button).toHaveClass("storybook-button--secondary");

    expect(component).toMatchSnapshot();
  });

  it("Should change size to small", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents<ButtonType>(Button, {
      label,
      size: "small",
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--small");
  });

  it("Should change size to large", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents<ButtonType>(Button, {
      label,
      size: "large",
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--large");
  });

  it("Should change size to primary", async () => {
    const label = "inner text";
    const { getByText } = renderWithEvents<ButtonType>(Button, {
      label,
      primary: true,
    });

    const button = getByText(label);
    expect(button).toHaveClass("storybook-button--primary");
  });

  it("Should trigger click", async () => {
    const label = "inner text";
    const handleClick = jest.fn();
    const { getByText } = renderWithEvents<ButtonType>(
      Button,
      { label },
      { click: handleClick }
    );

    const button = getByText(label);
    await fireEvent.click(button);

    expect(handleClick).toHaveBeenCalledTimes(1);
  });
});
