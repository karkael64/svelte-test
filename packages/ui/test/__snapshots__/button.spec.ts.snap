// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should instanciate with default values 1`] = `
Button {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "medium",
      "inner text",
      "storybook-button--secondary",
      [Function],
      false,
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "label": 1,
      "primary": 4,
      "size": 0,
    },
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
